app/models/workflow.rb

```diff
@@ -25,8 +25,18 @@ def external_signal?
     template&.signal_setting.present?
   end
 
+  def metadata
+    super.merge(:object_dependencies => object_dependencies)
+  end
+
   private
 
+  def object_dependencies
+    Hash.new { |h, k| h[k] = [] }.tap do |h|
+      tag_links.where(:workflow_id => id).pluck(:app_name, :object_type).uniq.each { |k, v| h[k] << v }
```

Comment:
@bzwei @hsong-rh Are there performance concerns with this running as part of an API  call returning a list of 50 or 100 workflows?  (UI caps returns at 100 right now.)

@eclarizio Thoughts on this?  I imagine we could look to use the ancillary_metadata approach here, but not sure we need to go that path just yet.

Reply:
@gmcculloug I removed the unnecessary `where` query. it does have impact on performance. But comparing with the other approach of adding separate endpoint to query these information, which means extra API calls, I feel this impact is acceptable.

Reply:
I would agree with that.  The table looks like it is indexed properly and we would not expect a lot of data here so probably would not see a that big of an impact overall.

Still curious if others have thoughts on this approach.  @eclarizio @mkanoor